### PR TITLE
Allow submitting AutoML Image job without a termination policy

### PR DESCRIPTION
**Pull request by rjaincc**
# Allow user to submit an AutoML image job without a termination policy
Fix for allowing user to submit an AutoML image job without a termination policy. 
Verified changes:
○ SDK: [ivory_prune_xzy005z2dy - Microsoft Azure Machine Learning Studio](https://ml.azure.com/experiments/id/3ad7f824-59d6-49db-93c2-15e17afe3c6b/runs/ivory_prune_xzy005z2dy?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
○ SDK: [plum_napa_m12vxtdbr8 - Microsoft Azure Machine Learning Studio](https://ml.azure.com/experiments/id/3ad7f824-59d6-49db-93c2-15e17afe3c6b/runs/plum_napa_m12vxtdbr8?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
○ CLI: [strong_jicama_yjsffn6d13 - Microsoft Azure Machine Learning Studio](https://ml.azure.com/experiments/id/3ad7f824-59d6-49db-93c2-15e17afe3c6b/runs/strong_jicama_yjsffn6d13?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
○ CLI: [serene_spoon_kx0z5cp9db - Microsoft Azure Machine Learning Studio](https://ml.azure.com/experiments/id/3ad7f824-59d6-49db-93c2-15e17afe3c6b/runs/serene_spoon_kx0z5cp9db?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
○ SDK/Pipeline: [automl_image_object_detection - Microsoft Azure Machine Learning Studio](https://ml.azure.com/experiments/id/3ad7f824-59d6-49db-93c2-15e17afe3c6b/runs/jolly_evening_nxcwly4xn9?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
○ CLI/Pipeline: [cli-pipeline-with-image-object-detection - Microsoft Azure Machine Learning Studio](https://ml.azure.com/experiments/id/3ad7f824-59d6-49db-93c2-15e17afe3c6b/runs/jovial_piano_4vypmkbj2s?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rupaljain-rs/providers/Microsoft.MachineLearningServices/workspaces/rupaljain-workspace&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by gauravrajguru — APPROVED** [review]

**Review by raduk — CHANGES_REQUESTED** [review]
1. Can you please add a unit test? We should strive to have a unit test for each bug (ideally it reproduces the bug then passes when it's corrected)
2. How about the CLI? Have you tested that as well?

**Comment by rjaincc** [comment]
> 1. Can you please add a unit test? We should strive to have a unit test for each bug (ideally it reproduces the bug then passes when it's corrected)
Valid point, unit tests added. 
> 2. How about the CLI? Have you tested that as well?
Yes. The jobs are already linked above.

**Review by gauravrajguru — CHANGES_REQUESTED** [review]
I'm little hesitant to add these unit test cases in automl jobs. I would prefer to test only image_sweep_settings class as a independent unit. (You can create a test file "test_automl_image_sweep_settings.py" for this class and add tests).

**Comment by rjaincc** [comment]
> I'm little hesitant to add these unit test cases in automl jobs. I would prefer to test only image_sweep_settings class as a independent unit. (You can create a test file "test_automl_image_sweep_settings.py" for this class and add tests).


@gauravrajguru  - Thanks for the discussion earlier, and the suggestion. Glad that we are on same page for this one. Updated the tests.

File: sdk/ml/azure-ai-ml/tests/automl_job/unittests/test_automl_image_sweep_setting.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# ---------------------------------------------------------
```
Comment (raduk) [inline sdk/ml/azure-ai-ml/tests/automl_job/unittests/test_automl_image_sweep_setting.py:1]: great to see this new file, I didn't like the duplication in the previous iteration :)

**Review by raduk — APPROVED** [review]

**Review by harneetvirk — APPROVED** [review]

**Pull request merged**